alibi_detect/utils/keops/kernels.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from pykeops.torch import LazyTensor
+import torch
+import torch.nn as nn
+from typing import Callable, Optional
+
+
+def sigma_mean(x: LazyTensor, y: LazyTensor, dist: LazyTensor) -> torch.Tensor:
+    """
+    Bandwidth estimation using the mean heuristic.
+
+    Parameters
+    ----------
+    x
+        LazyTensor of instances with dimension [Nx, 1, features].
+    y
+        LazyTensor of instances with dimension [1, Ny, features].
```

Comment:
Perhaps this is KeOps 101, but maybe worth adding here why there's an extra dimension for these tensors (compare to `torch` and `tensorflow` implementation without it. Perhaps link to https://www.kernel-operations.io/keops/_auto_tutorials/a_LazyTensors/plot_lazytensors_a.html ?

Reply:
Sure, can add that it's necessary to indicate how broadcasting takes place.